src/Microsoft.AspNetCore.Mvc.Analyzers/AddResponseTypeAttributeCodeFixAction.cs

```diff
@@ -0,0 +1,139 @@
+            foreach (var statusCode in statusCodes.OrderBy(s => s))
+            {
+                documentEditor.AddAttribute(context.MethodSyntax, CreateProducesResponseTypeAttribute(statusCode));
+            }
+
+            if (!declaredResponseMetadata.Any(m => m.IsDefault && m.AttributeSource == context.Method))
+            {
+                // Add a ProducesDefaultResponseTypeAttribute if the method does not already have one.
+                documentEditor.AddAttribute(context.MethodSyntax, CreateProducesDefaultResponseTypeAttribute());
+            }
+
+            return documentEditor.GetChangedDocument();
+        }
+
+        private async Task<CodeActionContext> CreateCodeActionContext(CancellationToken cancellationToken)
+        {
+            var root = await _document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await _document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var methodReturnStatement = (ReturnStatementSyntax)root.FindNode(_diagnostic.Location.SourceSpan);
+            var methodSyntax = methodReturnStatement.FirstAncestorOrSelf<MethodDeclarationSyntax>();
+            var method = semanticModel.GetDeclaredSymbol(methodSyntax, cancellationToken);
+
+            var symbolCache = new ApiControllerSymbolCache(semanticModel.Compilation);
+
+            var codeActionContext = new CodeActionContext(semanticModel, symbolCache, method, methodSyntax, cancellationToken);
+            return codeActionContext;
+        }
+
+        private ICollection<int> CalculateStatusCodesToApply(CodeActionContext context, IList<DeclaredApiResponseMetadata> declaredResponseMetadata)
+        {
+            if (!SymbolApiResponseMetadataProvider.TryGetActualResponseMetadata(context.SymbolCache, context.SemanticModel, context.MethodSyntax, context.CancellationToken, out var actualResponseMetadata))
+            {
+                // If we cannot parse metadata correctly, don't offer fixes.
+                return Array.Empty<int>();
+            }
+
+            var statusCodes = new HashSet<int>();
+            foreach (var metadata in actualResponseMetadata)
+            {
+                if (DeclaredApiResponseMetadata.TryGetDeclaredMetadata(declaredResponseMetadata, metadata, result: out var declaredMetadata) &&
+                    declaredMetadata.AttributeSource == context.Method)
+                {
+                    // A ProducesResponseType attribute is declared on the method for the current status code.
+                    continue;
+                }
+
+                statusCodes.Add(metadata.IsDefaultResponse ? 200 : metadata.StatusCode);
+            }
+
+            return statusCodes;
+        }
+
+        private static AttributeSyntax CreateProducesResponseTypeAttribute(int statusCode)
+        {
+            return SyntaxFactory.Attribute(
```

Comment:
`using static`?

Reply:
I used this in another version of this change - it's less typing, but I ended up typing `SyntaxFactory.` to look up what methods it offers far too often.